app/screens/browse_channels/channel_list.tsx

```diff
@@ -0,0 +1,137 @@
+import {useTheme} from '@context/theme';
+import {
+    changeOpacity,
+    makeStyleSheetFromTheme,
+} from '@utils/theme';
+
+import ChannelListRow from './channel_list_row';
+
+type Props = {
+    doGetChannels: () => void;
+    loading: boolean;
+    isSearch: boolean;
+    channels: Channel[];
+    onSelectChannel: (channel: Channel) => void;
+}
+
+const channelKeyExtractor = (channel: Channel) => {
+    return channel.id;
+};
+
+const getStyleFromTheme = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        noResultContainer: {
+            flexGrow: 1,
+            alignItems: 'center' as const,
+            justifyContent: 'center' as const,
+        },
+        noResultText: {
+            fontSize: 26,
+            color: changeOpacity(theme.centerChannelColor, 0.5),
+        },
+        loadingContainer: {
+            flex: 1,
+            justifyContent: 'center' as const,
+            alignItems: 'center' as const,
+        },
+        loading: {
+            height: 32,
+            width: 32,
+            justifyContent: 'center' as const,
+        },
+        listContainer: {
+            paddingHorizontal: 20,
+            flexGrow: 1,
+        },
+        separator: {
+            height: 1,
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.08),
+            width: '100%',
+        },
+    };
+});
+
+export default function ChannelList({
+    doGetChannels,
+    onSelectChannel,
+    loading,
+    isSearch,
+    channels,
+}: Props) {
+    const theme = useTheme();
+    const style = getStyleFromTheme(theme);
+
+    const renderItem = useCallback(({item}: {item: Channel}) => {
+        return (
+            <ChannelListRow
+                channel={item}
+                testID='browse_channels.custom_list.channel_item'
+                onPress={onSelectChannel}
+            />
+        );
+    }, [onSelectChannel]);
+
+    const renderLoading = useCallback(() => {
+        return (
+            <Loading
+                containerStyle={style.loadingContainer}
+                style={style.loading}
+                color={theme.buttonBg}
+            />
+        );
+
+    //Style is covered by the theme
+    }, [theme]);
+
+    const renderNoResults = useCallback(() => {
+        if (isSearch) {
+            return (
+                <View style={style.noResultContainer}>
+                    <FormattedText
+                        id='mobile.custom_list.no_results'
+                        defaultMessage='No Results'
+                        style={style.noResultText}
+                    />
+                </View>
+            );
+        }
+
+        return (
+            <View style={style.noResultContainer}>
+                <FormattedText
+                    id='browse_channels.noMore'
+                    defaultMessage='No more channels to join'
+                    style={style.noResultText}
+                />
+            </View>
+        );
```

Comment:
this is fine for now, but we should create a ticket to change the UI when there are no results in a way that matches the rest of the new UI.
cc: @Rina-dsg